docs/apis/plugintypes/blocks/_examples/block_pluginname.php

```diff
@@ -0,0 +1,50 @@
+class block_pluginname extends block_base {
+
+    /**
+     * Initialises the block.
+     *
+     * @return void
+     */
+    public function init() {
+        $this->title = get_string('pluginname', 'block_pluginname');
+    }
+
+    /**
+     * Gets the block contents.
+     *
+     *
+     * @return string The block HTML.
+     */
+    public function get_content() {
+        global $OUTPUT;
+
+        if ($this->content !== null) {
+            return $this->content;
+        }
+
+        $this->content = new stdClass();
+        $this->content->footer = '';
+
+        // Add logic here to define your template data or any other content.
+        $data = ['YOUR DATA GOES HERE'];
+
+        $this->content->text = $OUTPUT->render_from_template('block_yourplugin/content', $data);
+
+        return $this->content;
+    }
+
+    /**
+     * Defines in which pages this block can be added.
+     *
+     * @return array of the pages where the block can be added.
+     */
+    public function applicable_formats() {
+        return array(
+            'admin' => false,
+            'site-index' => true,
+            'course-view' => true,
+            'mod' => false,
+            'my' => true
```

Comment:
trailing comma please ;)

Reply:
Done. It was just old code copy-pasting. Thanks!